tool/python/singa/parameter.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+from utils.utility import * 
+from utils.message import * 
+from google.protobuf import text_format
+
+class Parameter(object):
+
+  def __init__(self, **kwargs):
+    '''
+    optional
+      **kwargs
+        name  = (string) // parameter name
+        lr    = (float)  // learning rate
+        wd    = (float)  // weight decay
+        init  = (string) // initialization type {'constant','uniform','gaussian'} 
+        value = (int)    // value for 'constant'
+        range = (float)  // [low, high] for 'uniform', low=-range, high=range
+        low   = (float)  // low value   for 'uniform'
+        high  = (float)  // high value  for 'uniform' 
+        mean  = (float)  // mean for 'gaussian'
+        std   = (float)  // std  for 'gaussian'
+    '''
+    fields = {'lr_scale' : kwargs['lr'] if 'lr' in kwargs else 1,
+              'wd_scale' : kwargs['wd'] if 'wd' in kwargs else 1
+             }
+    self.param = Message('Param', **fields).proto
+
+    if not 'name' in kwargs:
+      setval(self.param, name=generateName('param', 1))
+    else:
+      pname = kwargs['name']
+      if 'level' in kwargs:
+        pname += str(kwargs['level'])
+        if pname[0] == 'b':
+          pname += '2'
+      setval(self.param, name=pname)
+
+    if 'share_from' in kwargs:
+      setval(self.param, share_from=kwargs['share_from'])
+
+    # Param Generator
+    if 'range' in kwargs:
+      kwargs['low'] = -float(kwargs['range'])
+      kwargs['high'] = float(kwargs['range'])
+
+    if 'init' in kwargs:
+      if not kwargs['init'] == 'shared':
+        pg = Message('ParamGen', type=enumInitMethod(kwargs['init']), **kwargs)
+        del kwargs['init']
+        setval(self.param, init=pg.proto)
+    else: # default: uniform
+      pg = Message('ParamGen', type=enumInitMethod('uniform'))
+      setval(self.param, init=pg.proto)
+
+  def update(self, **fields):
+    '''
+    update parameter value
```

Comment:
Yes. We try to be consistent with Keira's. The we add arguments for supporting singa features.

Sent from my iPhone

> On 18 Dec 2015, at 6:01 PM, chonho <notifications@github.com> wrote:
> 
> In tool/python/singa/parameter.py:
> 
> > +    if 'range' in kwargs:
> > +      kwargs['low'] = -float(kwargs['range'])
> > +      kwargs['high'] = float(kwargs['range'])
> > +
> > +    if 'init' in kwargs:
> > +      if not kwargs['init'] == 'shared':
> > +        pg = Message('ParamGen', type=enumInitMethod(kwargs['init']), **kwargs)
> > +        del kwargs['init']
> > +        setval(self.param, init=pg.proto)
> > +    else: # default: uniform
> > +      pg = Message('ParamGen', type=enumInitMethod('uniform'))
> > +      setval(self.param, init=pg.proto)
> > +
> > +  def update(self, **fields):
> > +    '''
> > +    update parameter value
> Ok, I will set parameter initial values to be consistent with Keras
> But, after doing that, I think we need to explicitly set the value of most parameters to be consistent with Singa examples.
> Anyway, I will update it as soon as possible, and let's discuss this again.
> 
> —
> Reply to this email directly or view it on GitHub.
>